core/merginapi.cpp

```diff
@@ -2884,21 +2895,19 @@ void MerginApi::finishProjectSync( const QString &projectFullName, bool syncSucc
     QString projectNamespace, projectName;
     extractProjectName( projectFullName, projectNamespace, projectName );
     pushProject( projectNamespace, projectName );
+    // preserve information about GPKG schema changes detected at pull
+    mTransactionalStatus[projectFullName].gpkgSchemaChanged = hasSchemaChange;
```

Comment:
This can not be there. `[]` operator of QHash inserts a new element to the hashmap if there is none already.

We need to preserve this information differently - I suggest storing it in sync manager.